File: backend/internal/domain/models/activity.go
package models

import (
	"time"
)

type Activity struct {
	ID          int64     `json:"id"`
	Id_travel   int64     `json:"id_travel"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Location    string    `json:"location"`
	StartDate   time.Time `json:"start_date"`
	EndDate     time.Time `json:"end_date"`
	Price       float64   `json:"price"`
	Finished    bool      `json:"finished"`
}

File: frontend/types/activity.ts
const today = new Date();

export interface Activity {
	id: number;
	id_travel: number;
	name: string;
	description: string;
	location: string;
	start_date: string;
	end_date: string;
	price: number;
}

export const DEFAULT_ACTIVITY = {
	id: 0,
	id_travel: 0,
	name: '',
	description: '',
	location: '',
	start_date: today.getFullYear() + String(today.getMonth() + 1).padStart(2, '0') + String(today.getDate()).padStart(2, '0') + '',
	end_date: today.getFullYear() + String(today.getMonth() + 1).padStart(2, '0') + String(today.getDate()).padStart(2, '0') + '',
	price: 0,
}
